File: SCTE65/src/java/other/org/himalay/si/S_VCT_SubtableFactory.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.si;

import java.io.ByteArrayInputStream;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.util.ArrayList;
import org.himalay.msgs.runtime.Factory;
import org.himalay.msgs.runtime.FactoryProduct;
import org.himalay.msgs.runtime.Created;
import org.himalay.msgs.runtime.*;

@Created(date = "Sun Sep 14 22:48:37 EDT 2014")
public class S_VCT_SubtableFactory implements Factory {
	public static final int DCM_structure = 1;
	public static final int ICM_structure = 3;
	public static final int VCM_structure = 0;
	public static boolean isDebug = true;
	public static S_VCT_Subtable lastParsedObject = null;
	public static S_VCT_Subtable createMsg(int messageType,
			DataInputStream dis, IntegerHolder iHolder) throws IOException {
		int iRead = 0;
		S_VCT_Subtable retVal = null;
		try {
			switch (messageType) {
				case 1 : {
					retVal = new DCM_structure();
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 3 : {
					retVal = new ICM_structure();
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0 : {
					retVal = new VCM_structure();
					iRead += retVal.readNoHeader(dis);
					break;
				}
			}
		} catch (IOException ioex) {
			if (isDebug) {
				lastParsedObject = retVal;
			}
			throw ioex;
		}
		iHolder.setValue(iRead);
		return retVal;
	}

	public static abstract class S_VCT_Subtable extends BinMessage
			implements
				FactoryProduct {
		public S_VCT_Subtable() {
			super();
			// TODO Auto-generated constructor stub
		}

	}
}

// End of code

File: libsi/src/bp/S_VCT_SubtableFactory.cpp
//@Generated(date = "Mon Feb 01 22:29:56 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
#include "S_VCT_SubtableFactory.h"
#include "DCM_structure_base.h"
#include "DCM_structure.h"
#include "ICM_structure_base.h"
#include "ICM_structure.h"
#include "VCM_structure_base.h"
#include "VCM_structure.h"
// Namespace
namespace org {
namespace himalay {
namespace si {

using namespace ::org::himalay::msgs::runtime;


org::himalay::si::S_VCT_SubtableFactory::S_VCT_Subtable::S_VCT_Subtable() {};
org::himalay::si::S_VCT_SubtableFactory::S_VCT_Subtable::~S_VCT_Subtable() {};
bool S_VCT_SubtableFactory::isDebug = true;
/**
 *
 *S_VCT_SubtableFactory::S_VCT_SubtablePtr S_VCT_SubtableFactory::lastParsedObject= NULL;
 *
 */
org_himalay_si_S_VCT_SubtableFactory_S_VCT_Subtable__PTR_TYPE S_VCT_SubtableFactory::createMsg( int messageType, DataInputStream& istream, int& iHolder) throw(IOException)
{

    int retVal= 0;
    org_himalay_si_S_VCT_SubtableFactory_S_VCT_Subtable__PTR_TYPE retObj;
    try {
        switch( messageType )
        {
        case 1 :
        {
            retObj= org_himalay_si_S_VCT_SubtableFactory_S_VCT_Subtable__PTR_TYPE((S_VCT_SubtableFactory::S_VCT_Subtable *) new DCM_structure ());
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 3 :
        {
            retObj= org_himalay_si_S_VCT_SubtableFactory_S_VCT_Subtable__PTR_TYPE((S_VCT_SubtableFactory::S_VCT_Subtable *) new ICM_structure ());
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0 :
        {
            retObj= org_himalay_si_S_VCT_SubtableFactory_S_VCT_Subtable__PTR_TYPE((S_VCT_SubtableFactory::S_VCT_Subtable *) new VCM_structure ());
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        }
    } catch(IOException& ioex)
    {
        if (isDebug)
        {
//        lastParsedObject = retVal;
        }
        throw ioex;
    }
    iHolder=retVal;
    return retObj;
}
void intrusive_ptr_add_ref(org::himalay::si::S_VCT_SubtableFactory::S_VCT_Subtable* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::si::S_VCT_SubtableFactory::S_VCT_Subtable* p) {
    if (p->decreaseRef() == 0u) delete p;
};

}
}
} // End Namespace org.himalay.si




// End of code